frontend/choose-lang.php

```diff
@@ -298,15 +298,30 @@ public function parse_main_query( $query ) {
 			$this->set_language( $lang );
 			$this->set_curlang_in_query( $query );
 		} elseif ( ( count( $query->query ) == 1 || ( is_paged() && count( $query->query ) == 2 ) ) && $lang = get_query_var( 'lang' ) ) {
-			// Set is_home on translated home page when it displays posts. It must be true on page 2, 3... too.
 			$lang = $this->model->get_language( $lang );
 			$this->set_language( $lang ); // Set the language now otherwise it will be too late to filter sticky posts!
-			$query->is_home = true;
-			$query->is_tax = false;
+
+			// Set is_home on translated home page when it displays posts. It must be true on page 2, 3... too.
+			$query->is_home    = true;
+			$query->is_tax     = false;
 			$query->is_archive = false;
+
+			// Filters is_front_page() in case a static front page is not translated in this language.
```

Comment:
Would it be better to check if the front page is translated before filtering the option?
If I understood correctly, no need to filter it if the page is well translated, right?

Reply:
If the front page is translated, the part of the condtion is never reached.